service/extensions/extensions.go

```diff
@@ -72,6 +73,17 @@ func (bes *Extensions) NotifyPipelineNotReady() error {
 	return errs
 }
 
+func (bes *Extensions) NotifyConfig(ctx context.Context, conf *confmap.Conf) error {
```

Comment:
Can we check if `srv.collectorConf != nil` here so we can keep the `Settings.CollectorConf` optional?

Reply:
That's a good idea. I did this in `service.Start`.